src/moves.rs

```diff
@@ -44,11 +44,13 @@ impl<T: Clone + Copy + Default + PartialEq> ParallelMoves<T> {
     }
 
     fn sources_overlap_dests(&self) -> bool {
-        // Assumes `parallel_moves` has already been sorted in `resolve()` below.
-        for &(_, dst, _) in &self.parallel_moves {
+        // Assumes `parallel_moves` has already been sorted in
```

Comment:
can we clarify to "sorted by ..." for maximal explicitness here?

Reply:
Done, and rebased on top of #158 since you merged that. Like I contemplated in that PR but couldn't implement without also making this PR's changes, I've now also deferred checking overlaps until after deleting self-moves, since cycle resolution is not necessary for those.